src/net_processing.cpp

```diff
@@ -954,7 +1014,9 @@ void PeerManagerImpl::InitializeNode(CNode *pnode)
         assert(m_txrequest.Count(nodeid) == 0);
     }
     {
-        PeerRef peer = std::make_shared<Peer>(nodeid);
+        // Addr relay is disabled for outbound block-relay-only peers to
+        // prevent adversaries from inferring these links from addr traffic.
+        PeerRef peer = std::make_shared<Peer>(nodeid, /* addr_relay = */ !pnode->IsBlockOnlyConn());
```

Comment:
This logic of checking `IsBlockOnlyConn()` is used only when calling the `Peer` ctor. After that, we use the `RelayAddrsWithPeer()`, which replaces the old `RelayAddrsWithConn()` function. That's a clearer separation between net and net_processing since we only check the connection type when initializing the Peer object.